src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/EnumConverterTests.cs

```diff
@@ -554,5 +555,47 @@ public enum SampleEnumSByte : sbyte
             F = 1 << 5,
             G = 1 << 6,
         }
+
+        [Fact]
+        public static void HonorNamingPolicyOnDeserialization()
+        {
+            JsonSerializerOptions options = new()
+            {
+                Converters =
+                {
+                    new JsonStringEnumConverter(namingPolicy: new SimpleSnakeCasePolicy() )
+                }
+            };
+
+            BindingFlags bindingFlags = JsonSerializer.Deserialize<BindingFlags>(@"""non_public""", options);
+            Assert.Equal(BindingFlags.NonPublic, bindingFlags);
+
+            // Flags supported without naming policy.
+            bindingFlags = JsonSerializer.Deserialize<BindingFlags>(@"""NonPublic, Public""", options);
+            Assert.Equal(BindingFlags.NonPublic | BindingFlags.Public, bindingFlags);
+
+            // Flags not supported with naming policy.
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<BindingFlags>(@"""non_public, public""", options));
+
+            // Naming policy honored for values within the 64-count cache limit.
+            EnumWithMoreThan64Values customEnum = JsonSerializer.Deserialize<EnumWithMoreThan64Values>(@"""a_a_a""", options);
+            Assert.Equal(EnumWithMoreThan64Values.AAA, customEnum);
+
+            // Naming policy not honored for values outside the 64-count cache limit.
```

Comment:
Again, this seems odd to me. The cache limit might have impact on performance but never actually change the contract.